### DISCUSSION TITLE
how to draw additional traces in a network plotly plot

### DISCUSSION THREAD

**Discussion by AndreGismo** [e2nIEE/pandapower#1256] [Q&A]
Hello pandapower,

I would like to draw an additional trace of some colored  buses in an allready existing plotly plot of my network.

__Expected result:__
a plot with certain busses in a different color

__What I actually get:__
a separate plot containing only the specified buses (but nothing else)

__How I proceed:__
I use ``simple_plotly`` to plot the network.<br>As next I use ``create_bus_trace`` to specify those buses.<br>Finally I use ``draw_traces`` to plot the traces.

Here is, how my code looks like:

````
import numpy as np
import pandapower.plotting as pt
import pandapower as pp
import pandapower.networks as pn
import plotly
import plotly.graph_objs as go

net = pn.create_kerber_vorstadtnetz_kabel_1()
pt.simple_plotly(net)

how_much_buses = 33
color_buses = np.random.choice(net.bus.index, how_much_buses)
color_trace = pt.plotly.create_bus_trace(net, color_buses, color='red',
                                         trace_name='special buses')
pt.plotly.draw_traces(color_trace)
````

Is there a way to get the those colored buses into the network plot?

Many thanks for any help in advance!
Many greetings
André

**Comment by AndreGismo** [comment]
I have found a solution (I also had to change some settings in ``plotly.io`` for visualizing the updated plot):

here is the code, so it works as I have imagined:

````
import numpy as np
import pandapower.plotting as pt
import pandapower as pp
import pandapower.networks as pn
import plotly
import plotly.graph_objs as go
import plotly.graph_objs as go
import plotly.io as pio

net = pn.create_kerber_vorstadtnetz_kabel_1()
fig = pt.simple_plotly(net)

how_much_buses = 33
color_buses = np.random.choice(net.bus.index, how_much_buses)

fig.add_trace(go.Scatter(x=net.bus_geodata.loc[color_buses, 'x'],
                         y=net.bus_geodata.loc[color_buses, 'y'],
                         mode='markers'))

fig.show()
````
So the trick was to directly work with the ``plotly.graph_objs._figure.Figure`` object and manually adding another trace.
Another trick was to tell plotly to use the browser as standard renderer (otherwise the updated plot would not have been drawn - at least in my case working with spyder).

But I would still be interested, if there is a way to accomplish this directly with pandapower (without having to use those plotly functionalities).

Many greetings
André

**Reply by AndreGismo** [reply]
ups, the line telling plotly to use the browser as default renderer was missing:
``pio.renderers.default = 'browser'``

**Reply by rbolgaryn** [reply]
Hi @AndreGismo ,

thank you for the information! 

I work with plotly mainly in Jupyter Notebooks, or by using the browser. 

I think what you could do is write your own function instead of using sinple_plotly.

It could be useful to separate the simple_plotly function into the part that creates traces and the part that draws traces, so that the user can create the "basic" traces and add their own before calling the draw_traces function.
